express-notes-api/index.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+
+app.use(express.json());
+
+// GET
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  const notes = data.notes;
+  for (const property in notes) {
+    arr.push(notes[property]);
+  }
+  res.send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notes = data.notes;
+  if (isNaN(req.params.id) || req.params.id < 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  }
+  if (req.params.id === undefined) {
+    return res.status(404).send({ error: `cannot find note with id ${req.params.id}` });
+  }
+  for (const property in notes) {
+    if (property === req.params.id) {
+      res.status(200).send(notes[property]);
+    }
+  }
+});
+
+// POST
+
+app.post('/api/notes', (req, res) => {
+  if (req.body.content === undefined) {
+    res.status(400).send({ error: 'content is a required field' });
+  }
+  const newNote = { id: data.nextId, ...req.body };
+  data.notes = { ...data.notes, [data.nextId]: newNote };
+  data.nextId++;
+  if (!data.notes[newNote.id]) {
+    return res.status(500).send({ error: 'An unexpected error occurred.' });
+  }
+  res.status(201).send({ content: 'Inertia is a property of matter.' });
```

Comment:
You should not hardcode the `content` sent back, it should be the content of the new note that was created.